pkg/monitor/api/types.go

```diff
@@ -193,6 +195,32 @@ type AgentNotify struct {
 	Text string
 }
 
+// Decode decodes the message in 'data' into the struct.
+func (a *AgentNotify) Decode(data []byte) error {
+	buf := bytes.NewBuffer(data[1:])
+	dec := gob.NewDecoder(buf)
+	return dec.Decode(a)
+}
+
+// GetSrc retrieves the sorce endpoint for the message
+func (a *AgentNotify) GetSrc() uint16 {
+	return 0
+}
+
+// GetDst retrieves the destination endpoint for the message.
+func (a *AgentNotify) GetDst() uint16 {
+	return 0
+}
```

Comment:
Any reason to explicitly implement `GetSrc()` and `GetDst()` instead of embeding `DefaultSrcDstGetter` in `AgentNotify` if they always both return zero?

Is it because `(*AgentNotify).Decode` uses `gob.NewDecoder`? If correct, can you add a comment to explain this limitation please?